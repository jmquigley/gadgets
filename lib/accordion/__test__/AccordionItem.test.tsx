'use strict';

import * as assert from 'assert';
import {mount, shallow} from 'enzyme';
import * as React from 'react';
import {AccordionItem, getDefaultAccordionItemProps} from '../index';

test('Test retrieval of AccordionItem props object', () => {
	const props = getDefaultAccordionItemProps();

	assert(props);
	expect(props).toMatchSnapshot();
});

test('Test the creation of a AccordionItem control', () => {
	const ctl = shallow(
		<AccordionItem className="test-class" title="Test Title" />
	);

	assert(ctl);
	expect(ctl).toMatchSnapshot();
});

test('Test the disabling of the AccordionItem', () => {
	const ctl = shallow(
		<AccordionItem
			disabled
			title="Test Title"
		/>
	);

	assert(ctl);
	expect(ctl).toMatchSnapshot();
});

test('Test making the AccordionItem invisible', () => {
	const ctl = shallow(
		<AccordionItem
			title="Test Title"
			visible={false}
		/>
	);

	assert(ctl);
	expect(ctl).toMatchSnapshot();
});

test('Test clicking of the AccordionItem header', () => {
	const click = jest.fn();
	const child = <div>child content</div>;
	const ctl = mount(
		<AccordionItem
			initialToggle={false}
			onClick={click}
			title="Test Title"
		>
			{child}
		</AccordionItem>
	);

	assert(ctl);
	expect(ctl).toMatchSnapshot();

	assert(!ctl.state('toggle'));
	assert(!ctl.contains(child));
	ctl.find('.ui-label').simulate('click');
	expect(click).toHaveBeenCalled();
	assert(ctl.state('toggle'));
	assert(ctl.contains(child));
});
