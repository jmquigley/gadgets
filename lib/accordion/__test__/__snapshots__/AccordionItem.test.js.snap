// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the creation of a AccordionItem control 1`] = `
<div
  className="ui-accordionitem accordionItem test-class"
  style={Object {}}
>
  <Item
    backgroundColor="inherit"
    borderColor="inherit"
    borderWidth="none"
    className=""
    color="inherit"
    contentEditable={false}
    direction="right"
    disabled={false}
    hiddenLeftButton={false}
    hiddenRightButton={false}
    id=""
    layout={4}
    leftButton={null}
    location="none"
    noedit={false}
    noripple={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDoubleClick={[Function]}
    onKeyDown={[Function]}
    onKeyPress={[Function]}
    onMouseOut={[Function]}
    rightButton={
      <Button
        backgroundColor="inherit"
        borderColor="inherit"
        borderWidth="none"
        className=""
        color="inherit"
        contentEditable={false}
        direction="right"
        disabled={false}
        iconName="plus"
        iconStyle=""
        id=""
        location="none"
        noedit={false}
        noripple={false}
        onClick={[Function]}
        selected={false}
        sizing="normal"
        style={Object {}}
        visible={true}
      />
    }
    selected={false}
    sizing="normal"
    stacked={false}
    style={Object {}}
    title="Test Title"
    visible={true}
    widget={null}
  />
</div>
`;
